listenbrainz/webserver/views/api.py

```diff
@@ -177,6 +177,36 @@ def get_playing_now(user_name):
     })
 
 
+@api_bp.route("user/<user_name>/stats/artist")
+@crossdomain()
+@ratelimit()
+def get_artist(user_name):
+    """
+    Get top artists for user ``user_name``. The format for the JSON returned is defined in our :ref:`json-doc`.
+
+    **Note**: This endpoint is currently in beta and the JSON format may change in future
+
+    :statuscode 200: Successfull query
+    :statuscode 204: Statistics for user haven't been calculated, empty response will be returned
+    :statuscode 404: User not found
+    :resheader Content-Type: *application/json*
+    """
+
+    user = db_user.get_by_mb_id(user_name)
+    if user is None:
+        raise APINotFound("Cannot find user: %s" % user_name)
+
+    stats = db_stats.get_user_artists(user['id'])
+    if stats is None:
+        return '', 204
```

Comment:
We should always return valid json. Maybe return the same format but with an empty artist list?

Reply:
A 204 response does not include a message body and is terminated after first empty line after response headers.
https://restfulapi.net/http-status-204-no-content/